ocs_ci/utility/multicluster.py

```diff
@@ -0,0 +1,238 @@
+        "mco_upgrade",
+        "dr_hub_upgrade",
+        "acm_upgrade",
+        "pre_ocs_upgrade",
+        "ocs_upgrade",
+        "post_ocs_upgrade",
+        "post_upgrade",
+    ]
+
+    def __init__(self):
+        self.roles = []
+        # List of zones which are participating in this multicluster setup
+        self.zones = self.get_zone_info()
+        self.zone_base_rank = 100
+        # Each zone will be assigned with a rank
+        # This rank comes handy when we have to order the tests
+        self.zone_ranks = {}
+
+    def get_roles(self, metafunc):
+        """
+        should be overridden in the child class
+        Look for specific role markers based on multicluster scenario
+
+        Args:
+            metafunc: Pytest metafunc fixture object
+        """
+        pass
+
+    def generate_zone_ranks(self):
+        """
+        For each zone we would be generating the ranks, we will add the zone's respective indexes
+        to the base rank values which keeps the zone ranks apart and create spaces (for ranks)
+        in between to accomodate other tests
+
+        """
+        for i in range(len(self.zones)):
+            self.zone_ranks[f"{self.zones[i]}"] = (
+                self.zone_base_rank + i * self.zone_base_rank
+            )
+
+    def generate_role_ranks(self):
+        """
+        Based on the multicluster scenario, child class should generate the corresponding
+        role ranks. Roles are specific to multicluster scenarios
+
+        """
+        pass
+
+    def generate_pytest_parameters(self, metafunc, roles):
+        """
+        should be overridden in the child class.
+        This will be called for every testcase parametrization
+
+        """
+        pass
+
+    def get_zone_info(self):
+        """
+        Get the list of participating zones
+
+        """
+        zones = set()
+        for c in ocsci_config.clusters:
+            zones.add(c.ENV_DATA["zone"])
+        return list(zones)
+
+
+class MDRClusterUpgradeParametrize(MultiClusterUpgradeParametrize):
+    """
+    This child class handles MDR upgrade scenario specific pytest parametrization
+
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.roles_to_param_tuples = dict()
+        self.roles_to_config_index_map = dict()
+        self.zone_role_map = dict()
+        self.all_mdr_roles = MDR_ROLES
+
+        self.generate_zone_ranks()
+        self.generate_role_ranks()
+        self.generate_config_index_map()
+        # Reverse mapping of cluster's index to its role
+        self.index_to_role = {
+            index: role for role, index in self.roles_to_config_index_map.items()
+        }
+        self.generate_role_to_param_tuple_map()
+        self.generate_zone_role_map()
+
+    def generate_config_index_map(self):
+        """
+        Generate config indexes for all the MDRs cluster roles
+        ex: {"ActiveACM": 0, "PassiceACM": 2, "PrimaryODF": 1, "SecondaryODF": 3}
```

Comment:
```suggestion
        ex: {"ActiveACM": 0, "PassiveACM": 2, "PrimaryODF": 1, "SecondaryODF": 3}
```
Change all occurrences below as well